### PR TITLE
Fix uninstall dry-run has confirmation

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: operator/cmd/mesh/uninstall.go [hunk 1]
```diff
@@ -250,7 +250,7 @@ func preCheckWarnings(cmd *cobra.Command, kubeClient kube.CLIClient, uiArgs *uni
 			message += fmt.Sprintf(GatewaysRemovedWarning, gwList)
 		}
 	}
-	if uiArgs.skipConfirmation {
+	if dryRun || uiArgs.skipConfirmation {
```
Comment (ymesika) [inline operator/cmd/mesh/uninstall.go:253]: Is it really dry running? It seems like it will just print the message and continue with the run.

Comment (hanxiaop) [reply operator/cmd/mesh/uninstall.go:253]: Dryrun itself is functioning well. This is simply to bypass confirmation in `--dry-run` mode - which is no need in dryrun, similar to other istioctl commands.

Comment (ymesika) [reply operator/cmd/mesh/uninstall.go:253]: I see. Thanks.

**Review by ymesika — APPROVED** [review]
lgtm

**Review by zirain — APPROVED** [review]

**Pull request merged**